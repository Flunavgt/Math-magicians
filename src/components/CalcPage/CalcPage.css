.calcPageArea {
  width: 90vw;
  margin-inline: 5vw;
  display: flex;
}

.calcTitle {
  width: 50%;
  height: 70vh;
}

.calcTitle p {
  margin: 10%;
  font-size: 2.5rem;
  font-family: 'Monaco', sans-serif;
}

.CalcMain {
  margin: auto;
  display: flex;
  align-content: center;
}
