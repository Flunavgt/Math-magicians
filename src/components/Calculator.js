/* eslint-disable react/prefer-stateless-function */
/* eslint-disable arrow-body-style */
import React from 'react';
import '../Styles/style.css';

class Calculator extends React.Component {
  render() {
    return (
      <div className="calculatorMain">
        <p className="display">this is a test 006  </p>
        <KeyBoard />
      </div>

    );
  }
}

const KeyBoard = () => {
  return (
    <div className="KeyBoard">
      <button type="button" className="btnPad0">0</button>
      <button type="button" className="btnPad1">1</button>
      <button type="button" className="btnPad2">2</button>
      <button type="button" className="btnPad3">3</button>
      <button type="button" className="btnPad4">4</button>
      <button type="button" className="btnPad5">5</button>
      <button type="button" className="btnPad6">6</button>
      <button type="button" className="btnPad7">7</button>
      <button type="button" className="btnPad8">8</button>
      <button type="button" className="btnPad9">9</button>
      <button type="button" className="btnPadAdd">+</button>
      <button type="button" className="btnPadMinus">-</button>
      <button type="button" className="btnPadDiv">/</button>
      <button type="button" className="btnPadMultiply">X</button>
      <button type="button" className="btnPadEqual">=</button>
      <button type="button" className="btnPadPercentage">%</button>
      <button type="button" className="btnPadAllClear">AC</button>
      <button type="button" className="btnPadMasMin">+/-</button>
      <button type="button" className="btnPadDecimal">.</button>
    </div>

  );
};
export default Calculator;
