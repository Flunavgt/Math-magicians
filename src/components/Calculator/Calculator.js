import { useState } from 'react';
import '../../Styles/style.css';
import calculate from '../../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({ total: '', next: '', operation: '' });

  const handleClick = (event) => {
    const value = event.target.textContent;
    const solution = calculate(state, value);
    setState(solution);
  };

  const { total, next, operation } = state;
  return (
    <div className="calculatorMain">
      <p className="display" data-testid="display">
        { total }
        { operation }
        { next }
      </p>
      <KeyBoard handler={handleClick} />
    </div>

  );
};

const KeyBoard = (props) => {
  const { handler } = props;
  return (
    <div className="KeyBoard">
      <button type="button" onClick={handler} className="btnPad0">0</button>
      <button type="button" onClick={handler} className="btnPad1">1</button>
      <button type="button" onClick={handler} className="btnPad2">2</button>
      <button type="button" onClick={handler} className="btnPad3">3</button>
      <button type="button" onClick={handler} className="btnPad4">4</button>
      <button type="button" onClick={handler} className="btnPad5">5</button>
      <button type="button" onClick={handler} className="btnPad6">6</button>
      <button type="button" onClick={handler} className="btnPad7">7</button>
      <button type="button" onClick={handler} className="btnPad8">8</button>
      <button type="button" onClick={handler} className="btnPad9">9</button>
      <button type="button" onClick={handler} className="btnPadAdd">+</button>
      <button type="button" onClick={handler} className="btnPadMinus">-</button>
      <button type="button" onClick={handler} className="btnPadDiv">รท</button>
      <button type="button" onClick={handler} className="btnPadMultiply">x</button>
      <button type="button" onClick={handler} className="btnPadEqual">=</button>
      <button type="button" onClick={handler} className="btnPadPercentage">%</button>
      <button type="button" onClick={handler} className="btnPadAllClear">AC</button>
      <button type="button" onClick={handler} className="btnPadMasMin">+/-</button>
      <button type="button" onClick={handler} className="btnPadDecimal">.</button>
    </div>

  );
};
export default Calculator;
