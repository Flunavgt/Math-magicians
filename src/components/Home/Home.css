.homeTxt {
  width: 80vw;
  margin-inline: 10vw;
  font-family: 'Monaco', sans-serif;
}
