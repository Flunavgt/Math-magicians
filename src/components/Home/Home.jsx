import React from 'react';

export default function Home() {
  return (
    <div>
      <div>Welcome to our Page</div>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia,
        molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum
        numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium
        optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis
        obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam
        nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit,
        tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit,
        quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos
        sapiente officiis modi at sunt excepturi expedita sint? Sed quibusdam
        recusandae alias error harum maxime adipisci amet laborum. Perspiciatis
      </p>
    </div>
  );
}
