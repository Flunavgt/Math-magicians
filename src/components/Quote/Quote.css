.quoteMathMag {
  width: 90vw;
  margin-inline: 5vw;
  height: 85vh;
}

.quoteMathMag p {
  text-align: center;
  margin-top: 30%;
  font-size: 1.5rem;
  margin-inline: 5%;
  font-family: 'Monaco', sans-serif;
}
