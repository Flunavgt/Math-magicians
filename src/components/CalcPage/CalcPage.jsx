import React from 'react';
import Calculator from '../Calculator/Calculator';
import './CalcPage.css';

export default function CalcPage() {
  return (
    <div className="calcPageArea">
      <div className="calcTitle">
        <p>Let&apos;s do some Math!</p>
      </div>
      <div className="CalcMain">
        <Calculator />
      </div>
    </div>
  );
}
