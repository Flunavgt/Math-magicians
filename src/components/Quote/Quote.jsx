import React from 'react';
import './Quote.css';

export default function Quote() {
  return (
    <div className="quoteMathMag">
      <p>
        Mathematics is bit about numbers, equations, computations,
        or algoriths: it is about understanding.
        - William Paul Thurston
      </p>
    </div>
  );
}
