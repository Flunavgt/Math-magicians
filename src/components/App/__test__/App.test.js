import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import CalcPage from '../../CalcPage/CalcPage';
import Calculator from '../../Calculator/Calculator';
import Home from '../../Home/Home';
import Quote from '../../Quote/Quote';
import App from '../App';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText('Welcome to our Page');
  expect(linkElement).toBeInTheDocument();
});

test('snapshot', () => {
  const tree = render(<Home />);
  expect(tree).toMatchSnapshot();
});

test('Quote', () => {
  const tree = render(<Quote />);
  expect(tree).toMatchSnapshot();
});

test('Calc page', () => {
  const tree = render(<CalcPage />);
  expect(tree).toMatchSnapshot();
});

test('Calculator', () => {
  render(<Calculator />);
  const btn = screen.getByText('2');
  userEvent.click(btn);
  expect(screen.getByTestId('display')).toHaveTextContent('2');
});
