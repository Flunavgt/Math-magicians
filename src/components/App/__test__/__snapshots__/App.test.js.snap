// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calc page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calcPageArea"
      >
        <div
          class="calcTitle"
        >
          <p>
            Let's do some Math!
          </p>
        </div>
        <div
          class="CalcMain"
        >
          <div
            class="calculatorMain"
          >
            <p
              class="display"
              data-testid="display"
            />
            <div
              class="KeyBoard"
            >
              <button
                class="btnPad0"
                type="button"
              >
                0
              </button>
              <button
                class="btnPad1"
                type="button"
              >
                1
              </button>
              <button
                class="btnPad2"
                type="button"
              >
                2
              </button>
              <button
                class="btnPad3"
                type="button"
              >
                3
              </button>
              <button
                class="btnPad4"
                type="button"
              >
                4
              </button>
              <button
                class="btnPad5"
                type="button"
              >
                5
              </button>
              <button
                class="btnPad6"
                type="button"
              >
                6
              </button>
              <button
                class="btnPad7"
                type="button"
              >
                7
              </button>
              <button
                class="btnPad8"
                type="button"
              >
                8
              </button>
              <button
                class="btnPad9"
                type="button"
              >
                9
              </button>
              <button
                class="btnPadAdd"
                type="button"
              >
                +
              </button>
              <button
                class="btnPadMinus"
                type="button"
              >
                -
              </button>
              <button
                class="btnPadDiv"
                type="button"
              >
                รท
              </button>
              <button
                class="btnPadMultiply"
                type="button"
              >
                x
              </button>
              <button
                class="btnPadEqual"
                type="button"
              >
                =
              </button>
              <button
                class="btnPadPercentage"
                type="button"
              >
                %
              </button>
              <button
                class="btnPadAllClear"
                type="button"
              >
                AC
              </button>
              <button
                class="btnPadMasMin"
                type="button"
              >
                +/-
              </button>
              <button
                class="btnPadDecimal"
                type="button"
              >
                .
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calcPageArea"
    >
      <div
        class="calcTitle"
      >
        <p>
          Let's do some Math!
        </p>
      </div>
      <div
        class="CalcMain"
      >
        <div
          class="calculatorMain"
        >
          <p
            class="display"
            data-testid="display"
          />
          <div
            class="KeyBoard"
          >
            <button
              class="btnPad0"
              type="button"
            >
              0
            </button>
            <button
              class="btnPad1"
              type="button"
            >
              1
            </button>
            <button
              class="btnPad2"
              type="button"
            >
              2
            </button>
            <button
              class="btnPad3"
              type="button"
            >
              3
            </button>
            <button
              class="btnPad4"
              type="button"
            >
              4
            </button>
            <button
              class="btnPad5"
              type="button"
            >
              5
            </button>
            <button
              class="btnPad6"
              type="button"
            >
              6
            </button>
            <button
              class="btnPad7"
              type="button"
            >
              7
            </button>
            <button
              class="btnPad8"
              type="button"
            >
              8
            </button>
            <button
              class="btnPad9"
              type="button"
            >
              9
            </button>
            <button
              class="btnPadAdd"
              type="button"
            >
              +
            </button>
            <button
              class="btnPadMinus"
              type="button"
            >
              -
            </button>
            <button
              class="btnPadDiv"
              type="button"
            >
              รท
            </button>
            <button
              class="btnPadMultiply"
              type="button"
            >
              x
            </button>
            <button
              class="btnPadEqual"
              type="button"
            >
              =
            </button>
            <button
              class="btnPadPercentage"
              type="button"
            >
              %
            </button>
            <button
              class="btnPadAllClear"
              type="button"
            >
              AC
            </button>
            <button
              class="btnPadMasMin"
              type="button"
            >
              +/-
            </button>
            <button
              class="btnPadDecimal"
              type="button"
            >
              .
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Quote 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="quoteMathMag"
      >
        <p>
          Mathematics is bit about numbers, equations, computations, or algoriths: it is about understanding. - William Paul Thurston
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="quoteMathMag"
    >
      <p>
        Mathematics is bit about numbers, equations, computations, or algoriths: it is about understanding. - William Paul Thurston
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="homeTxt"
      >
        <div>
          <h2>
            Welcome to our Page
          </h2>
        </div>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit, quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos sapiente officiis modi at sunt excepturi expedita sint? Sed quibusdam recusandae alias error harum maxime adipisci amet laborum. Perspiciatis
        </p>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit, quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos sapiente officiis modi at sunt excepturi expedita sint? Sed quibusdam recusandae alias error harum maxime adipisci amet laborum. Perspiciatis
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="homeTxt"
    >
      <div>
        <h2>
          Welcome to our Page
        </h2>
      </div>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit, quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos sapiente officiis modi at sunt excepturi expedita sint? Sed quibusdam recusandae alias error harum maxime adipisci amet laborum. Perspiciatis
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit, quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos sapiente officiis modi at sunt excepturi expedita sint? Sed quibusdam recusandae alias error harum maxime adipisci amet laborum. Perspiciatis
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
